src/shadow.rs

```diff
@@ -335,8 +335,8 @@ struct ContainsSelf {
 }
 
 impl<'v> Visitor<'v> for ContainsSelf {
-    fn visit_ident(&mut self, _: Span, ident: Ident) {
-        if self.name == ident.unhygienic_name {
+    fn visit_name(&mut self, _: Span, name: Name) {
+        if self.name == name {
```

Comment:
> One more missing unhygienize() here.

Stupid me. That lint seems ok now. Thanks.

> they need to be moved to AST

I would have done that but `used_underscore_binding` needs to get the parent of an expression. With `hir` that’s in `rustc::hir::map::Map`, there does not seem to be an `ast` equivalent. I’ll see if we can rewrite the lint to not use that.